packages/headless/src/features/attached-results/attached-results-slice.ts

```diff
@@ -0,0 +1,54 @@
+import {isNullOrUndefined} from '@coveo/bueno';
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  setAttachedResults,
+  attachResult,
+  detachResult,
+} from './attached-results-actions';
+import {
+  AttachedResult,
+  getAttachedResultsInitialState,
+} from './attached-results-state';
+
+export const attachedResultsReducer = createReducer(
+  getAttachedResultsInitialState(),
+  (builder) => {
+    builder
+      .addCase(setAttachedResults, (state, action) => {
+        const {results: attachedResults, message} = action.payload;
+
+        if ('attachedResults' in state && state['results']?.length > 0) {
+          return;
+        }
+
+        state['results'] = attachedResults;
+        state['message'] = message;
+      })
```

Comment:
why not use isNullOrUndefined since it's imported?

Reply:
Sorry I don't get where you'd want me to use the isNullOrUndefined ?